integration_tests/test_builtin_str.py

```diff
@@ -63,6 +63,15 @@ def test_str_slice_step():
     assert s[::-1] == "kjihgfedcba"
 
 
+def test_issue_883():
+    s: str
+    s = "abcde"
+    c: str
+    for c in s[::-1]:
+        print(c)
```

Comment:
This looks okay. Do we support `reversed` intrinsic as well?

Reply:
Not yet.

Reply:
Okay. Makes sense. Let's support that in a subsequent PR.